src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -2456,7 +2471,14 @@ public void execute() {
 
    public void focus()
    {
-      view_.editorContainer().focus();
+      if (isVisualModeActivated())
```

Comment:
`isVisualModeActivated()` is a logical state (should it be active) whereas the actual load state (and presence of the `panmirror_` member variable that you access in `focus()`) is different (i.e. it could be logically "activated" but still in the process of loading, so `panmirror_` is `null`). I think you probably want to wrap this call in `ensureVisualModeActive()`.

Reply:
Actually, I don't even know if that would be correct. Here is another effort I made to get focus working when a visual mode document is opened as part of navigation: https://github.com/rstudio/rstudio/blob/66cec22c35f0eee23e53d908ae93d1975076d007/src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/visualmode/VisualMode.java#L628-L640

Reply:
I think we should talk realtime about all of this on Monday. I'll send a schedule item.

Reply:
Would it make sense to first check `isActivated()` and then next `ensureVisualModeActivated()` so the command is properly registered if we are indeed still loading?